src/views/Home/components/CakeHarvestBalance.tsx

```diff
@@ -39,7 +39,7 @@ const CakeHarvestBalance = () => {
   return (
     <Block>
       <CardValue value={earningsSum} lineHeight="1.5" />
-      {!cakePriceBusd.eq(0) && <CardBusdValue value={earningsBusd} />}
+      {cakePriceBusd.gt(0) && <CardBusdValue value={earningsBusd} />}
```

Comment:
I understand why these `!....eq(0)` > `.gt(0)` conditions are an improvement and directly address the small issue that the PR is highlighting - but curious as to whether the other conditional changes are fixing _actual_ problems?

Particularly when tackling issues that appear momentarily in specific circumstances, it would be good if you could include a short video/instructions on how to recreate the issue - it eases the burden on us figuring what's wrong/how your work fixes it.

Reply:
> That fix actually fixes the issue that shown in the image I put in description.

Yea I realise that. My comment is more pointed towards the other changes.

> It shows ~$0 when cake price is being loaded and in Firefox (strangely) after the cake price is loaded. And more strangely it only happens in production build not the development one. 

All very important information - browser & env specific instructions on how to recreate it. Please include this in the PR description.

> happens because now we have NaN instead of 0 at first

Yea it'll be undefined now right? Maybe an alternative solution to look at at some point would be initialising the `busdPrice` as a zero BN.

Reply:
Don't want to touch the new usePriceCakeBusd logic, but if we add there to return BIG_ZERO when output is NaN we don't need to do all these changes. What do you prefer?

Reply:
I think it would make sense to ensure we're always going to get a BN returned by those hooks (`usePriceBnbBusd` & `useBusdPriceFromPid` too).

Is the root cause that the farm isn't yet in state? i.e. `cakeBnbFarm` is undefined?
or is the issue that the `token` doesn't yet have a `busdPrice`, so `token.busdPrice` is undefined? 

If it's the latter I think it'll be as simple as `return new BigNumber(cakeBnbFarm.token.busdPrice || 0)`?

I think you're still right to modify the conditions that I mentioned in my first comment, from  `!....eq(0)` > `.gt(0)`

With regards to whether the other changes will still be necessary - I'll leave that up to you.

Reply:
Is the root cause that the farm isn't yet in state? i.e. cakeBnbFarm is undefined?
or is the issue that the token doesn't yet have a busdPrice, so token.busdPrice is undefined?

`cakeBnbFarm is defined but doesn't yet have its public farm data or token.busdprice (due to both farms (cake- bnb or bnb-busd) promises might get stuck which i try to fix it here https://github.com/pancakeswap/pancake-frontend/pull/1318) `